daemon/src/maker_rollover.rs

```diff
@@ -0,0 +1,245 @@
+                })
+            }),
+            receiver,
+            (self.oracle_pk, announcement),
+            RolloverParams::new(
+                cfd.order.price,
+                cfd.quantity_usd,
+                cfd.order.leverage,
+                cfd.refund_timelock_in_blocks(),
+            ),
+            Role::Maker,
+            dlc,
+            self.n_payouts,
+        );
+
+        let this = ctx.address().expect("acquired own actor address");
+
+        // tell maker_cfd_actor that we are finishing doing the rollover
+        spawn_fallible::<_, anyhow::Error>(async move {
+            let _ = match contract_future.await {
+                Ok(dlc) => {
+                    this.send(RollOverCompleted::Success { order_id, dlc })
+                        .await?
+                }
+                Err(error) => {
+                    this.send(RollOverCompleted::Failed { order_id, error })
+                        .await?
+                }
+            };
+            Ok(())
+        });
+
+        Ok(())
+    }
+
+    pub async fn handle_reject_rollover(
+        &mut self,
+        msg: RejectRollOver,
+        ctx: &mut xtra::Context<Self>,
+    ) -> Result<()> {
+        self.send_to_taker_actor
+            .send(TakerMessage {
+                taker_id: self.taker_id,
+                msg: MakerToTaker::RejectRollOver(msg.order_id),
+            })
+            .await??;
+        ctx.stop();
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl Handler<AcceptRollOver> for Actor {
+    async fn handle(&mut self, msg: AcceptRollOver, ctx: &mut Context<Self>) {
+        log_error!(self.handle_accept_rollover(msg, ctx));
+    }
+}
+
+#[async_trait]
+impl Handler<RejectRollOver> for Actor {
+    async fn handle(&mut self, msg: RejectRollOver, ctx: &mut Context<Self>) {
+        log_error!(self.handle_reject_rollover(msg, ctx));
+    }
+}
+
+#[async_trait]
+impl Handler<RollOverMsg> for Actor {
+    async fn handle(&mut self, msg: RollOverMsg, _ctx: &mut Context<Self>) {
+        log_error!(self.handle_rollover_msg(msg));
+    }
+}
+
+#[xtra_productivity]
+impl Actor {
+    async fn handle_rollover_completed(
+        &mut self,
+        msg: RollOverCompleted,
+        ctx: &mut Context<Self>,
+    ) -> Result<()> {
+        match msg {
+            RollOverCompleted::Success { .. } => {
+                self.maker_cfd_actor.send(msg).await??;
+                ctx.stop();
+            }
+            RollOverCompleted::Rejected { order_id } => {
+                tracing::info!(%order_id, "Rollover rejected");
+                self.maker_cfd_actor.send(msg).await??;
+                ctx.stop();
+            }
+            RollOverCompleted::Failed {
+                ref order_id,
+                ref error,
+            } => {
+                tracing::error!(%order_id, "Rollover failed: {}", error);
+                self.maker_cfd_actor.send(msg).await??;
+                ctx.stop();
+            }
+        }
```

Comment:
This seems unnecessarily repetitive?

```suggestion
	self.maker_cfd_actor.send(msg).await??;
        ctx.stop();
```

Note that "logging AND passing on an error" is almost always undesired and will result in duplicate log messages.